src/main/java/io/vertx/gsoc2023/qotd/QoTDHandlers.java

```diff
@@ -23,36 +34,52 @@ public void getAllQuotes(RoutingContext ctx) {
         .end(quotes.encode()));
   }
 
+  public void postNewQuote(RoutingContext context) {
+    var payload = context.body().asJsonObject();
+    insertQuote(payload)
+      .onFailure(cause -> context.response().setStatusCode(400).end())
+      .onSuccess(quote -> {
+        eventBus.send(QUOTE_INSERTED, quote);
+        context.response().setStatusCode(201).end();
+      });
+  }
+
+  public void realtimeQuotes(RoutingContext context) {
+    context.request()
+      .toWebSocket()
+      .onSuccess(wsServer -> {
```

Comment:
Can you propose a change to handle the error case?

Reply:
I did it, although I could not at this time come with a way to test